jax/_src/lib/xla_bridge.py

```diff
@@ -257,6 +257,22 @@ def make_gpu_client(*, platform_name, visible_devices_flag):
 if iree is not None:
   register_backend_factory("iree", iree.iree_client_factory, priority=-100)
 
+# Register PJRT C API plugins based on environment variables.
+def _initialize_plugin_backends():
+  def make_factory(name, path):
+    def factory():
+      xla_client._xla.load_pjrt_plugin(name, path)
+      return xla_client._xla.get_c_api_client(name)
```

Comment:
@skye Do we want to expose xla_client._xla in xla_bridge?

Reply:
Ideally no, but it's not the end of the world. The right way to do this is to update [xla_client.py](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/python/xla_client.py) in the TF repo to expose the methods you need directly (like [this](https://github.com/tensorflow/tensorflow/blob/6d5d88041f49e866b1dd18b6f532107e4b84e650/tensorflow/compiler/xla/python/xla_client.py#L41-L42)). @stellaraccident I'd help you get a TF PR in if you wanna do that, but this is also fine for now and we can fix it later (might be easiest to do one big pass exposing everything we need after this code stops churning).

I suspect this does need an xla_client version guard though. Basically, this code has to be forward-compatible with older jaxlib versions (and xla_client is shipped with jaxlib).

Example version guard: https://github.com/google/jax/blob/e3c2602ed5174bcfd79fe3b25cf9009cdcb48bfe/jax/experimental/host_callback.py#L993-L997

Version number comes from here: https://github.com/tensorflow/tensorflow/blob/6d5d88041f49e866b1dd18b6f532107e4b84e650/tensorflow/compiler/xla/python/xla_client.py#L46

You can drop the guard once the minimum jaxlib version exports a high enough xla_client version:
https://github.com/google/jax/blob/e3c2602ed5174bcfd79fe3b25cf9009cdcb48bfe/jax/version.py#L19

So for `get_c_api_client`, blame tells me it was introduced in https://github.com/tensorflow/tensorflow/commit/77a1e8e3fc3637f7fa29d9cd8d8d33dcb707b965, and the xla_client version number was [110](https://github.com/tensorflow/tensorflow/blob/77a1e8e3fc3637f7fa29d9cd8d8d33dcb707b965/tensorflow/compiler/xla/python/xla_client.py#L46). Or you can just use the current version number of 118 if you don't wanna check. I'm gonna cut a new jaxlib soon, so this should be publicly available then.